src/panels/lovelace/editor/hui-yaml-card-preview.ts

```diff
@@ -16,29 +15,27 @@ export class HuiYAMLCardPreview extends HTMLElement {
     }
   }
 
-  set yaml(value: string) {
+  set config(config: LovelaceConfig) {
     if (this.lastChild) {
       this.removeChild(this.lastChild);
     }
 
-    if (value === "") {
+    if (!config) {
       return;
     }
 
-    let conf;
-    try {
-      conf = yaml.safeLoad(value);
-    } catch (err) {
-      conf = createErrorCardConfig(`Invalid YAML: ${err.message}`, undefined);
-    }
-
-    const element = createCardElement(conf);
+    const element = createCardElement(config);
 
     if (this._hass) {
       element.hass = this._hass;
     }
 
     this.appendChild(element);
+    this._config = config;
+  }
+
+  get config() {
```

Comment:
If we would want this, we could just store it as `this.config` directly, however I think that the use case is wrong and we should keep the value to save in the dialog editor card.

Reply:
The reason for this is that it tightly couples the implementation of the preview card with the edit dialog and limits reusability of the preview card.